src/XMakeBuildEngine/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -320,6 +325,7 @@ private BuildParameters(BuildParameters other)
             _disableInProcNode = other._disableInProcNode;
             _logTaskInputs = other._logTaskInputs;
             _logInitialPropertiesAndItems = other._logInitialPropertiesAndItems;
+            _warningsAsErrors = other._warningsAsErrors == null ? null : new HashSet<string>(other._warningsAsErrors, StringComparer.OrdinalIgnoreCase);
```

Comment:
AFAIK it's highly dependent on the individual use case, so we probably can't tell for this case without extensive benchmarking.

My understanding is based on [Pike's third rule](http://users.ece.utexas.edu/~adnan/pike.html): Fancy algorithms are slow when n is small, and n is usually small. In this case, a hashtable is awesomely super fast, linear lookup . . . but requires hashing and indirection/pointer-following. A sorted set that is small, as I expect this thing to be, could easily use fewer instructions per lookup.

But we'd have to measure to know (Rule 2!) and I don't think this will be worth worrying about, so  I'd say to leave it as-is.